### PR TITLE
NAS-112217 / 21.10 / The WebDav Services/config icon disappears in the Sharing -> WebDav screen after choosing Config Service

### PR DESCRIPTION
**Pull request by AlexKarpov98**

### CODE REVIEW

File: src/app/pages/sharing/components/shares-dashboard/shares-dashboard.component.html [hunk 1]
```diff
@@ -4,24 +4,68 @@
...
+      </mat-menu>
+    </ng-container>
   </app-expandable-table>
   <app-expandable-table [ngClass]="getIscsiOrder()" [expandableTableState]="iscsiExpandableState" [disabled]="true" [conf]="iscsiTableConf">
     <ng-container
       headerCustomContent
       *ngTemplateOutlet="iscsiServiceStatus === ServiceStatus.Loading ? statusLoading : statusButton; context: { status: iscsiServiceStatus, count: iscsiHasItems }"
     ></ng-container>
+    <ng-container *ngIf="iscsiTableConf?.tableExtraActions" headerCustomActions>
+      <button mat-icon-button [matMenuTriggerFor]="rowActions" aria-label="Table Actions of Expandable Table">
```
Comment (denysbutenko) [inline src/app/pages/sharing/components/shares-dashboard/shares-dashboard.component.html:59]: Seems like `headerCustomActions` has a similar template across tables. Can we extract it to a common template and use ngTemplateOutlet with context to pass values?

Comment (AlexKarpov98) [reply src/app/pages/sharing/components/shares-dashboard/shares-dashboard.component.html:59]: Fixed.

**Comment by denysbutenko** [comment]
Steps to reproduce:

- Create/access a WebDav share
- Click on the three dots on the right to Config Service
- Click Cancel
- The Screen goes to Services

Getting this error on Services page
```ERROR TypeError: Cannot read properties of undefined (reading 'length')
    at t.get isSingleAction [as isSingleAction] (entity-table-actions.component.ts:29)
    at template (entity-table-actions.component.html:2)```

Also, there is an issue when loading the Shares page:
![image](https://user-images.githubusercontent.com/351613/133658815-3799da10-3df3-454a-b6c1-bedd9441b9bf.png)

**Comment by denysbutenko** [comment]
Probably it would be nice to get a user back to the Shares page if he comes from Shares. Maybe in another ticket

**Comment by AlexKarpov98** [comment]
@denysbutenko 
Error on Services page is not from my commit, maybe others latest update breaks it.
Also regarding invisible config menu issue on Shared Dashboard loading, we need to wait for api response to get service status and the labels on config menu depend on it.

**Comment by denysbutenko** [comment]
@AlexKarpov98, that's true about the Services page, it is broken on the master branch as well.
For the screenshot above I mean missing padding on the right. Feel free to check how it is working currently.

**Review by dszidi — COMMENTED** [review]
Running into the same problem with the services page that Denys noticed. FWIW the fix is simple...

image

**Comment by denysbutenko** [comment]
Bug fix for the Services page was merged to the master branch.

**Comment by AlexKarpov98** [comment]
> @AlexKarpov98, that's true about the Services page, it is broken on the master branch as well.
> For the screenshot above I mean missing padding on the right. Feel free to check how it is working currently.

Fixed

**Review by denysbutenko — APPROVED** [review]
Looks good!

**Pull request merged**